expirationd.lua

```diff
@@ -166,6 +166,8 @@ local function worker_loop(task)
             end
         end
 
+        --check if fiber was canceled in pcall
+        fiber.testcancel()
```

Comment:
Yes, you are right, I reviewed how these checks are arranged and redid it a bit. I also left comments on each check, about why it should be here. Please look and let's discuss what you think is the best thing to do?

Reply:
To be honest it looks a bit tricky. Ideally each line should be covered by test. But I guess it's not so.
Anyway "Check if fiber was canceled" doesn't explain why above we use "task.worker_canceled" but here we use testcancel?

Reply:
We use both test_cancel and task.worker_canceled here.
Check this out, is this a good description?
```diff
@@ -177,11 +177,15 @@ local function worker_loop(task)
             end
         end

-        -- Check if fiber was canceled.
-        fiber.testcancel()
-        -- Then there is sleep and it can be long.
+        -- Check cancellation of fiber,
+        -- then there is sleep and it can be long.
         if task.worker_canceled then
+            -- The worker_canceled flag is used only if we have atomic_iteration = true.
             fiber.self():cancel()
+        else
+            -- Check if fiber was canceled, because pcall
+            -- can catch a cancellation and can skip it.
+            fiber.testcancel()
         end
         -- Full scan iteration is complete, yield
         fiber.sleep(task.full_scan_delay)
```

Reply:
> -- Check if fiber was canceled, because pcall
            -- can catch a cancellation and can skip it.
            fiber.testcancel()

Who cancelled this fiber? And why it doesn't switch worker_canceled flag to true?

Reply:
> Who canceled this fiber? And why it doesn't switch worker_canceled flag to true?  
 
We can write about the task:stop logic, but why describe what you can see in the code?
I missed your point, what do you want to see here as comments?

Reply:
Seems I told about this part:
https://github.com/tarantool/expirationd/blob/930c1c4852cb6eea6ac32bb758753ad6d9bd71f3/expirationd.lua#L215-L219

It's a bit inconsistent. I thought that stop() turn worker_canceled flag on anyway. But seems it's relevant only for atomic_iteration.

I don't know will anyone support me, but I vote for setup `worker_canceled = true` for not batch_iteration case as well. Seems it also should solve an issue.

Reply:
Yes, this is a good option, then we don't need testcancel check at all. Perhaps it is worth doing so. @ligurio can help discuss it as a third person

Reply:
Personally I like a variant with `worker_canceled = true` for non-`batch_iteration` as well.

Reply:
Finished with this. Changed worker_canceled not only for atomic and removed testcancel.